src/main/java/org/mockito/internal/matchers/CompareTo.java

```diff
@@ -18,12 +18,8 @@ public CompareTo(Comparable<T> value) {
     }
 
     @SuppressWarnings("unchecked")
-    public boolean matches(Object actual) {
-        if(!(actual instanceof Comparable)) {
-            return false;
-        }
-        
-        return matchResult(((Comparable) actual).compareTo(wanted));
+    public boolean matches(T actual) {
+        return matchResult(((Comparable)actual).compareTo(wanted));
```

Comment:
`<T extends Comparable<T>>` and I've modified the function to have an input parameter of type `T`, so the cast check is not necessary.